service/src/main/java/org/mskcc/cmo/metadb/service/impl/MessageHandlingServiceImpl.java

```diff
@@ -128,20 +128,17 @@ public void run() {
                                 oldCmoPatientId, newCmoPatientId);
 
                         for (MetadbSample sample: samples) {
-                            // TODO: add support for clinical sample updates
+                            SampleMetadata latestMetadata = sample.getLatestSampleMetadata();
+                            latestMetadata.setCmoPatientId(newCmoPatientId);
                             if (sample.getSampleCategory().equals("research")) {
-                                SampleMetadata latestMetadata = sample.getLatestSampleMetadata();
-                                latestMetadata.setCmoPatientId(newCmoPatientId);
-                                String newCmoSampleLabel = latestMetadata.getCmoSampleName()
-                                        .replaceAll(oldCmoPatientId, newCmoPatientId);
-
                                 LOG.info("Updating patient ID prefix embedded in CMO sample label "
                                         + "for research sample: " + latestMetadata.getPrimaryId());
+                                String newCmoSampleLabel = latestMetadata.getCmoSampleName()
+                                        .replaceAll(oldCmoPatientId, newCmoPatientId);
                                 latestMetadata.setCmoSampleName(newCmoSampleLabel);
-                                sample.updateSampleMetadata(latestMetadata);
-                            } else if (sample.getSampleCategory().equals("clinical")) {
-                                LOG.info("CLINICAL SAMPLE UPDATES NOT SUPPORTED YET");
                             }
+                            sample.setPatient(updatedPatient);
```

Comment:
Updated patient is returned above https://github.com/mskcc/cmo-metadb/pull/586/files#diff-e1e08df99e843f0281d446ab742c0cb98fdbd395eb2a5904c73059dc827ce160R127

We just needed to update the associated `Patient` object of the research and/or clinical sample whose cmo patient id is also being updated